### ISSUE TITLE
devon python pip not working

### ISSUE THREAD

**Issue by hohwille** [devonfw/ide#839]
It seems that with my last minute changes in PR #807 I broke something about pip as @alfeilex pointed out in #824.
To make it short:
```
devon python pip
D:\projects\devonfw\software\python\python.exe: can't open file 'D:\\projects\\devonfw\\software\\pip': [Errno 2] No such file or directory
```

I am sorry for this and will fix it on Monday.

**Comment by hohwille** [comment]
It might even make sense to add an extra commandlet for `pip` in devonfw-ide.
See discussion in #824.
This would make the `python` commandlet simpler and avoid hacks like moving pip files from `Scripts` to the actual python installation.
However, I still have to see how python and pip works on other platforms and if all this makes sense and gives a consistent result.
It was a lot easier for me to integrate tools I already knew from a users perspective like `java` or `mvn`.
Recently we are integrating a lot of tools without knowing the internal design of these tools and the people we are doing that for are not in touch with us to tell whether A or B is actually the better way to go.

**Comment by hohwille** [comment]
The problem is that pip is an executable itself and the `python` commandlet is calling pip via python what is IMHO wrong.
You can call pip via python but then you indeed have to provide the complete path to the executable to python what is kind of odd:
```
$ devon python pip
D:\projects\devonfw\software\python\python.exe: can't open file 'D:\\projects\\devonfw\\software\\pip': [Errno 2] No such file or directory
$ devon python python/pip
D:\projects\devonfw\software\python\python.exe: can't open file 'D:\\projects\\devonfw\\software\\python\\pip': [Errno 2] No such file or directory
$ devon python python/pip.exe

Usage:
  pip <command> [options]
...
```

This confirms that it is required to add a separate commandlet `pip` and move the pip logic out of python.

**Comment by hohwille** [comment]
Python and pip remain a big problem:
```
 pip --version
Traceback (most recent call last):
  File "runpy.py", line 196, in _run_module_as_main
  File "runpy.py", line 86, in _run_code
  File "D:\projects\devonfw\software\python\pip.exe\__main__.py", line 4, in <module>
ModuleNotFoundError: No module named 'pip'
```

**Comment by hohwille** [comment]
```
$ python --version
D:/projects/devonfw/software/python/bin/python.exe: error while loading shared libraries: ?: cannot open shared object file: No such file or directory
```

**Comment by hohwille** [comment]
```
Python path configuration:
  PYTHONHOME = (not set)
  PYTHONPATH = 'D:\projects\devonfw\software\vscode-extensions;D:\projects\devonfw\software\vscode\bin;D:\projects\devonfw\software\sonarqube\bin;D:\projects\devonfw\software\python;D:\projects\devonfw\software\pip\bin;D:\projects\devonfw\software\node;D:\projects\devonfw\software\maven\bin;D:\projects\devonfw\software\java\bin;D:\projects\devonfw\software\helm;D:\projects\devonfw\software\eclipse;D:\projects\devonfw\software\dotnet;D:\projects\devonfw\software\cobigen-cli\bin;D:\projects\devonfw\software\az;C:\Users\hohwille\.sdkman\candidates\java\current\bin;C:\Users\hohwille\.sdkman\candidates\gradle\current\bin;C:\Users\hohwille\.sdkman\candidates\cxf\current\bin;C:\Users\hohwille\bin;C:\Program Files\Git\usr\local\bin;C:\Program Files\Git\usr\bin;C:\Program Files\Git\usr\bin;C:\Program Files\Git\opt\bin;C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin;C:\Program Files\AdoptOpenJDK\jdk-15.0.0.36-hotspot\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0;C:\WINDOWS\System32\OpenSSH;C:\Program Files (x86)\GnuPG\bin;C:\Program Files\SlikSvn\bin;C:\Program Files\Git\cmd;C:\Users\hohwille\scripts;C:\Program Files\chocolatey;C:\Program Files\chocolatey\lib\shellcheck\content;C:\Program Files (x86)\GnuWin32\bin;C:\Users\hohwille\.dotnet\tools;C:\Users\hohwille\AppData\Local\Programs\Rancher Desktop\resources\resources\win32\bin;C:\Users\hohwille\AppData\Local\Programs\Rancher Desktop\resources\resources\linux\bin;C:\Program Files\Git\usr\bin\vendor_perl;C:\Program Files\Git\usr\bin\core_perl'
  program name = 'D:\projects\devonfw\software\python\bin\python.exe'
  isolated = 0
  environment = 1
  user site = 1
  import site = 1
  sys._base_executable = 'D:\\projects\\devonfw\\software\\python\\bin\\python.exe'
  sys.base_prefix = ''
  sys.base_exec_prefix = ''
  sys.platlibdir = 'lib'
  sys.executable = 'D:\\projects\\devonfw\\software\\python\\bin\\python.exe'
  sys.prefix = ''
  sys.exec_prefix = ''
  sys.path = [
    'D:\\projects\\devonfw\\software\\vscode-extensions',
    'D:\\projects\\devonfw\\software\\vscode\\bin',
    'D:\\projects\\devonfw\\software\\sonarqube\\bin',
    'D:\\projects\\devonfw\\software\\python',
    'D:\\projects\\devonfw\\software\\pip\\bin',
    'D:\\projects\\devonfw\\software\\node',
    'D:\\projects\\devonfw\\software\\maven\\bin',
    'D:\\projects\\devonfw\\software\\java\\bin',
    'D:\\projects\\devonfw\\software\\helm',
    'D:\\projects\\devonfw\\software\\eclipse',
    'D:\\projects\\devonfw\\software\\dotnet',
    'D:\\projects\\devonfw\\software\\cobigen-cli\\bin',
    'D:\\projects\\devonfw\\software\\az',
    'C:\\Users\\hohwille\\.sdkman\\candidates\\java\\current\\bin',
    'C:\\Users\\hohwille\\.sdkman\\candidates\\gradle\\current\\bin',
    'C:\\Users\\hohwille\\.sdkman\\candidates\\cxf\\current\\bin',
    'C:\\Users\\hohwille\\bin',
    'C:\\Program Files\\Git\\usr\\local\\bin',
    'C:\\Program Files\\Git\\usr\\bin',
    'C:\\Program Files\\Git\\usr\\bin',
    'C:\\Program Files\\Git\\opt\\bin',
    'C:\\Program Files (x86)\\Microsoft SDKs\\Azure\\CLI2\\wbin',
    'C:\\Program Files\\AdoptOpenJDK\\jdk-15.0.0.36-hotspot\\bin',
    'C:\\WINDOWS\\system32',
    'C:\\WINDOWS',
    'C:\\WINDOWS\\System32\\Wbem',
    'C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0',
    'C:\\WINDOWS\\System32\\OpenSSH',
    'C:\\Program Files (x86)\\GnuPG\\bin',
    'C:\\Program Files\\SlikSvn\\bin',
    'C:\\Program Files\\Git\\cmd',
    'C:\\Users\\hohwille\\scripts',
    'C:\\Program Files\\chocolatey',
    'C:\\Program Files\\chocolatey\\lib\\shellcheck\\content',
    'C:\\Program Files (x86)\\GnuWin32\\bin',
    'C:\\Users\\hohwille\\.dotnet\\tools',
    'C:\\Users\\hohwille\\AppData\\Local\\Programs\\Rancher Desktop\\resources\\resources\\win32\\bin',
    'C:\\Users\\hohwille\\AppData\\Local\\Programs\\Rancher Desktop\\resources\\resources\\linux\\bin',
    'C:\\Program Files\\Git\\usr\\bin\\vendor_perl',
    'C:\\Program Files\\Git\\usr\\bin\\core_perl',
    'D:\\projects\\devonfw\\software\\python\\bin\\python310.zip',
    'D:\\projects\\devonfw\\software\\python\\bin',
  ]
Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding
Python runtime state: core initialized
ModuleNotFoundError: No module named 'encodings'

Current thread 0x0000b9a8 (most recent call first):
  <no Python frame>
```

**Comment by hohwille** [comment]
Python on windows is fighting like a beast against following a standardized structure with a `bin` subfolder containing the executables. I do not know how we can tame this beast called Python.
Is it so hard to create an installation in `software/python` with the executables in `software/python/bin` that is actually working?

**Comment by hohwille** [comment]
fixed with PR #841

**Comment by hohwille** [comment]
For the record: https://justinmayer.com/posts/homebrew-python-is-not-for-you/
